### PR TITLE
Added Dijkstra Algorithm document

### PR DESCRIPTION
**Pull request by fuboki10**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [ ] I fixed #your-issue-number-here

### Code Snippets

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### Documentation

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language> Article` format
- [ ] I followed the applicable article template
  - [ ] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [ ] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [ ] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

### Testing

- [ ] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [ ] I followed the testinfo template, if applicable

## Notes

Feel free to describe what you added or updated.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[3]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/CODE_ARTICLE_TEMPLATE.md
[4]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/LANGUAGE_ARTICLE_TEMPLATE.md
[5]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/PROJECT_ARTICLE_TEMPLATE.md

### CODE REVIEW

File: docs/projects/dijkstra/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Example
+
+Dijkstra's algorithm to find the shortest path between a and b. It picks the unvisited vertex with the low distance, calculates the distance through it to each unvisited neighbor, and updates the neighbor's distance if smaller. Mark visited (set to red) when done with neighbors.
+
+![Dijkstra_Animation](https://user-images.githubusercontent.com/35429211/67672949-a2dcfd80-f981-11e9-862a-96bd0ec9ba83.gif)
+
+## Requirements
+
+```console
+./dijkstra.lang "4, 3, 2, 1, 3, 1, 3, 1, 3, 4, 5, 2, 3"
```
Comment (jrg94) [inline docs/projects/dijkstra/index.md]: I like the setup here, but I think I'd prefer to break out the inputs, so they're more distinct:

```suggestion
./dijkstra.lang 4 3 "2, 1, 3, 1, 3, 1, 3, 4, 5" 2 3
```

Order is fine.

Comment (fuboki10) [reply docs/projects/dijkstra/index.md]: I think that's better too

**Comment by fuboki10** [comment]
Matrix is good for small inputs I think it's better than adj list in this case.
I can also remove (Source, Destination) and the output will be all Minimum Costs from a fixed source like node (0) for example :

| Vertex | Distance |
| ---------- | ----------- |
| 0 | 0 |
| 1 | 4 |
| 2 | 12 |
| 3 | 10 |

What about this ?

**Comment by jrg94** [comment]
This is great! Although, part of me still wants to be able to select a source (and possibly a destination). That way, the adjacency matrix can be reused, and we don't have to create a new one with ordered nodes. 

At this point, you've put a lot of work into this, so I'll accept pretty much whatever you like.

File: docs/projects/dijkstra/index.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+| Mapping | 0   | 1   | 2   | 3   | 4   |
+| ------- | --- | --- | --- | --- | --- |
+| 0       | 0   | 2   | 0   | 6   | 0   |
+| 1       | 2   | 0   | 3   | 8   | 5   |
+| 2       | 0   | 3   | 0   | 0   | 7   |
+| 3       | 6   | 8   | 0   | 0   | 9   |
+| 4       | 0   | 5   | 7   | 9   | 0   |
+
+Then we take the Source and the Destination
```
Comment (jrg94) [inline docs/projects/dijkstra/index.md]: Is part of this description missing? Would also be a good idea to talk about the output.

File: docs/projects/dijkstra/index.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Then we take the Source and the Destination 
+
+
+## Testing
+
+| Description      | Matrix   | Source      | Destination                                                                     | Output                                                     |
+| ---------------- | -------- | ------------| ----------- | ----------------------------------------------- |
+| No Input         |          |             |             |                                         | "Usage: please provide a comma-separated list of integers" |
+| Empty Input      | ""       |      ""       |       ""     |                                          | "Usage: please provide a comma-separated list of integers" |
+| Non-Square Input | "1, 0, 3, 0, 5, 1" |      "1"    |  "2"                                                     | "Usage: please provide a comma-separated list of integers" |
```
Comment (jrg94) [inline docs/projects/dijkstra/index.md]: I'd like to expand the test cases a bit here, so we know we get proper testing.

File: docs/projects/dijkstra/index.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+| 4       | 0   | 5   | 7   | 9   | 0   |
+
+Then we take the Source and the Destination 
+
+
+## Testing
+
+| Description      | Matrix   | Source      | Destination                                                                     | Output                                                     |
+| ---------------- | -------- | ------------| ----------- | ----------------------------------------------- |
+| No Input         |          |             |             |                                         | "Usage: please provide a comma-separated list of integers" |
```
Comment (jrg94) [inline docs/projects/dijkstra/index.md]:
```suggestion
| No Input         |          |             |                                         | "Usage: please provide a comma-separated list of integers" |
```

I think there's one too many of these.

File: docs/projects/dijkstra/index.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+| Empty Input      | ""       |      ""       |       ""     |                                          | "Usage: please provide a comma-separated list of integers" |
```
Comment (jrg94) [inline docs/projects/dijkstra/index.md]:
```suggestion
| Empty Input      | ""       |      ""       |       ""     |  "Usage: please provide a comma-separated list of integers" |
```

Same issue here.

File: docs/projects/dijkstra/index.md (hunk 3 above)
Comment (fuboki10) [reply docs/projects/dijkstra/index.md]: Ok i will think about some cases

**Comment by jrg94** [comment]
Looks great!

**Comment by jrg94** [comment]
Whoops! One last request. Are you in the author file? https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/_data/authors.yml

**Comment by fuboki10** [comment]
Thank you, added my self in Author file.

**Pull request merged**